scenes/game/local_game.gd

```diff
@@ -5560,6 +5837,154 @@ func strike_send_attack_to_discard_or_gauge(performing_player : Player, card):
+				if hit:
+					active_strike.extra_attack_hit = true
+					active_strike.extra_attack_state = ExtraAttackState.ExtraAttackState_Hit
+					active_strike.extra_attack_remaining_effects = get_all_effects_for_timing("hit", attacker_player, attacker_card, true, true)
+				else:
+					active_strike.extra_attack_state = ExtraAttackState.ExtraAttackState_After
+					active_strike.extra_attack_remaining_effects = get_all_effects_for_timing("after", attacker_player, attacker_card, true, true)
+			ExtraAttackState.ExtraAttackState_Hit:
+				events += do_remaining_effects(attacker_player, ExtraAttackState.ExtraAttackState_Hit_Response)
+			ExtraAttackState.ExtraAttackState_Hit_Response:
+				var incoming_damage = calculate_damage(attacker_player, defender_player)
+				events += do_hit_response_effects(attacker_player, defender_player, incoming_damage, ExtraAttackState.ExtraAttackState_Hit_ApplyDamage)
+			ExtraAttackState.ExtraAttackState_Hit_ApplyDamage:
+				events += apply_damage(attacker_player, defender_player)
+				active_strike.extra_attack_state = ExtraAttackState.ExtraAttackState_After
+				active_strike.extra_attack_remaining_effects = get_all_effects_for_timing("after", attacker_player, attacker_card, true, true)
+				if game_over:
+					active_strike.strike_state = ExtraAttackState.ExtraAttackState_Cleanup
+			ExtraAttackState.ExtraAttackState_After:
+				events += do_remaining_effects(attacker_player, ExtraAttackState.ExtraAttackState_Cleanup)
+			ExtraAttackState.ExtraAttackState_Cleanup:
+				active_strike.extra_attack_state = ExtraAttackState.ExtraAttackState_CleanupEffects
+				active_strike.extra_attack_remaining_effects = get_all_effects_for_timing("cleanup", attacker_player, attacker_card, true, true)
+			ExtraAttackState.ExtraAttackState_CleanupEffects:
+				events += do_remaining_effects(attacker_player, ExtraAttackState.ExtraAttackState_Complete)
+			ExtraAttackState.ExtraAttackState_Complete:
+				var card_name = attacker_card.definition['display_name']
+				if active_strike.extra_attack_hit:
+					_append_log_full(Enums.LogType.LogType_CardInfo, attacker_player, "adds their extra attack %s to gauge." % card_name)
+					events += attacker_player.add_to_gauge(attacker_card)
+				else:
+					_append_log_full(Enums.LogType.LogType_CardInfo, attacker_player, "discards their extra attack %s." % card_name)
+					events += attacker_player.add_to_discards(attacker_card)
+
+				# Finally, resume the original strike.
+				active_strike.extra_attack_in_progress = false
+				if game_over:
+					change_game_state(Enums.GameState.GameState_GameOver)
+				else:
+					# Intentional events = because events are passed in.
+					events = continue_resolve_strike(events)
+				break
+	return events
+
+func determine_if_attack_hits(events, attacker_player : Player, defender_player : Player, card : GameCard):
+	# events actually is used, but there's an unused parameter warning for some reason, so add _.
```

Comment:
is this an outdated comment?

Reply:
yes, this was me being confused about a warning, which was actually a real bug.
So, if you use events += [] inside this function, this does a new assignment and loses the reference to events and doesn't actually add to the array.
So if we ever aren't returning events from a function we pass events into, it needs to use append_array/append.
Potentially we should go change this everywhere anyway.